packages/mco/components/protected-applications/list-page.tsx

```diff
@@ -0,0 +1,339 @@
+import {
+  Button,
+  ButtonVariant,
+  Pagination,
+  PaginationVariant,
+  Text,
+  TextVariants,
+  Flex,
+  FlexItem,
+  Tabs,
+  Tab,
+  Tooltip,
+} from '@patternfly/react-core';
+import { InProgressIcon, CubeIcon } from '@patternfly/react-icons';
+import { ActionsColumn, Td, Tr } from '@patternfly/react-table';
+import { ODFMCO_OPERATOR_NAMESPACE } from '../../constants';
+import {
+  getDRPlacementControlResourceObj,
+  getDRPolicyResourceObj,
+} from '../../hooks';
+import { DRPlacementControlModel } from '../../models';
+import { DRPlacementControlKind, DRPolicyKind } from '../../types';
+import { getLastAppDeploymentClusterName, getDRPolicyName } from '../../utils';
+import {
+  EmptyRowMessage,
+  NoDataMessage,
+  AlertMessages,
+  getHeaderColumns,
+  getColumnNames,
+  getRowActions,
+  APP_DETAILS,
+  getReplicationHealth,
+  SyncStatus,
+  SyncStatusInfo,
+  getErrorConditions,
+  isFailingOrRelocating,
+  getAppWorstStatus,
+  AppHealthMap,
+  ExpandableComponentsProps,
+  ExpandableComponentsMap,
+} from './helper';
+
+const INITIAL_PAGE_NUMBER = 1;
+const COUNT_PER_PAGE_NUMBER = 10;
+
+type RowExtraProps = {
+  launcher: LaunchModal;
+  navigate: NavigateFunction;
+  syncStatus: SyncStatus;
+};
+
+const ProtectedAppsTableRow: React.FC<
+  RowComponentType<DRPlacementControlKind>
+> = ({ row: application, rowIndex, extraProps }) => {
+  const { t } = useCustomTranslation();
+  const [appDetails, setAppDetails] = React.useState(APP_DETAILS.DEFAULT);
+
+  const columnNames = getColumnNames(t);
+  const appName = getName(application);
+  const { launcher, navigate, syncStatus }: RowExtraProps = extraProps;
+
+  const onTabSelect = (
+    _event: React.MouseEvent<HTMLElement, MouseEvent>,
+    tabKey: APP_DETAILS
+  ) =>
+    tabKey === appDetails
+      ? // collapse the expandable section
+        setAppDetails(APP_DETAILS.DEFAULT)
+      : // render new selected expandable section
+        setAppDetails(tabKey);
+
+  const isExpanded: boolean = appDetails !== APP_DETAILS.DEFAULT;
+  const enrolledNamespaces: string[] =
+    // ToDo: Update with correct spec field which will report all protected namespaces
+    // @ts-ignore
+    application.spec?.enrolledNamespace || [];
+  const filteredConditions: K8sResourceCondition[] =
+    getErrorConditions(application);
+  const showErrorsDetails: boolean =
+    !!filteredConditions.length || appDetails === APP_DETAILS.ERRORS;
+  const errorsCount: number = filteredConditions.length;
+  const showEventsDetails: boolean =
+    isFailingOrRelocating(application) || appDetails === APP_DETAILS.EVENTS;
+  const eventsCount: number = isFailingOrRelocating(application) ? 1 : 0;
+  const syncStatusInfo: SyncStatusInfo =
+    syncStatus[appName] || ({} as SyncStatusInfo);
+  const { icon, title }: AppHealthMap = getAppWorstStatus(syncStatusInfo, t);
+  const ExpandableComponent: React.FC<ExpandableComponentsProps> =
+    ExpandableComponentsMap[appDetails];
```

Comment:
```suggestion
  // Namespace info
  const enrolledNamespaces: string[] =
    // ToDo: Update with correct spec field which will report all protected namespaces
    // @ts-ignore
    application.spec?.enrolledNamespace || [];
    
    // Error info
  const filteredConditions: K8sResourceCondition[] =
    getErrorConditions(application);
  const showErrorsDetails: boolean =
    !!filteredConditions.length || appDetails === APP_DETAILS.ERRORS;
  const errorsCount: number = filteredConditions.length;
  
  // Current activity info
  const showEventsDetails: boolean =
    isFailingOrRelocating(application) || appDetails === APP_DETAILS.EVENTS;
  const eventsCount: number = isFailingOrRelocating(application) ? 1 : 0;
  
  // Replication health info
  const syncStatusInfo: SyncStatusInfo =
    syncStatus[appName] || ({} as SyncStatusInfo);
  const { icon, title }: AppHealthMap = getAppWorstStatus(syncStatusInfo, t);
  
  // Expandable section
  const ExpandableComponent: React.FC<ExpandableComponentsProps> =
    ExpandableComponentsMap[appDetails];
```

optional comment for better code readability